app/controllers/application_controller.rb

```diff
@@ -187,8 +188,11 @@ def expanded_embed(url, caption=nil)
       end
     end
 
-    render_to_string partial: "/articles/embeds/#{slug}.html.erb", locals: { embed_id: embed_id || url, caption: caption }
+    render_to_string partial: "/articles/embeds/#{slug}.html.erb", locals: { embed_id: embed_id || url, caption: caption, link: link }
   end
   helper_method :expanded_embed
 
+  def url_or_path?(string)
+    string.match?(/^(http|\/)\S+/)
```

Comment:
What cases does this cover?

- https URLs
- http URLs
- /relative/to/site/root/paths
- relative/to/document/paths

The last one seem least important and probably the hardest to account for.